eth/backend.go

```diff
@@ -972,6 +972,10 @@ func New(ctx context.Context, stack *node.Node, config *ethconfig.Config, logger
 			return nil, errors.New("nil sentryClient for polygon sync")
 		}
 
+		var openDatabase heimdall.OpenDatabaseFunc = func(ctx context.Context, name string, logger log.Logger) (kv.RwDB, error) {
+			return node.OpenDatabase(ctx, stack.Config(), kv.ConsensusDB, name, false, logger)
```

Comment:
can heimdall not simply call `node.OpenDatabase` internally to avoid passing in this as an external dependency?
to be it feels like this should be an internal responsibility of the heimdall component

Reply:
tried that and seems like there is a circular dependency that needs to be solved